gui/mozregui/global_prefs.py

```diff
@@ -54,6 +55,29 @@ def __init__(self, parent=None):
         self.ui.http_timeout.setValue(options['http_timeout'])
         self.ui.persist_size_limit.setValue(options['persist_size_limit'])
         self.ui.bg_downloads.setChecked(options['background_downloads'])
+        self.ui.advanced_options.setText(QString("Show Advanced Options"))
+        self.hide_adv_options()
+        self.ui.advanced_options.clicked.connect(self.toggle_adv_options)
+
+    def toggle_adv_options(self):
+        if self.ui.advanced_options.text() == QString("Show Advanced Options"):
+            self.ui.advanced_options.setText(QString("Hide Advanced Options"))
+            self.show_adv_options()
+        else:
+            self.ui.advanced_options.setText(QString("Show Advanced Options"))
+            self.hide_adv_options()
+
+    def show_adv_options(self):
```

Comment:
Could you rewrite this using setVisible(bool) instead of show()/hide() ? You will only need one function then to hide or show the items instead of two.